File: encodify/Classes/Encode/Controller/DecodeViewController.m
//
//  DecodeViewController.m
//  encodify
//
//  Created by 洪朔 on 16/8/31.
//  Copyright © 2016年 Tuccuay. All rights reserved.
//

#import "DecodeViewController.h"

#import "EncodifyBase64.h"
#import "EncodifyUnicode.h"
#import "EncodifyMorse.h"
#import "EncodifyURI.h"

@interface DecodeViewController ()

@end

@implementation DecodeViewController

- (NSString *)titleForPagerTabStripViewController:(XLPagerTabStripViewController *)pagerTabStripViewController {
    return @"Decode";
}

- (NSString *)encodeWithBase64:(NSString *)inputString {
    return [EncodifyBase64 decodeByBase64WithString:inputString];
}

- (NSString *)encodeWithUnicode:(NSString *)inputString {
    return [EncodifyUnicode decodeByUnicodeStringWIthString:inputString];
}

- (NSString *)encodeWithMorse:(NSString *)inputString {
    return [EncodifyMorse decodeByMorseWithString:inputString];
}

- (NSString *)encodeWithURI:(NSString *)inputString {
    return [EncodifyURI decodeByURIWithString:inputString];
}

- (NSString *)encodeButtonTitle {
    return @"Decode";
}

@end

File: encodify/Classes/Encode/Controllers/DecodeViewController.swift
//
//  DecodeViewController.swift
//  encodify
//
//  Created by 洪朔 on 2024/12/20.
//  Copyright © 2024年 Tuccuay. All rights reserved.
//

import UIKit
import XLPagerTabStrip

class DecodeViewController: EncodeBaseViewController, IndicatorInfoProvider {
    
    // MARK: - IndicatorInfoProvider
    nonisolated func indicatorInfo(for pagerTabStripController: PagerTabStripViewController) -> IndicatorInfo {
        return IndicatorInfo(title: "Decode")
    }
    
    // MARK: - Override encoding methods
    override func encodeWithBase64(_ inputString: String) -> String? {
        return Base64Encoder.decode(inputString)
    }
    
    override func encodeWithUnicode(_ inputString: String) -> String? {
        return UnicodeEncoder.decode(inputString)
    }
    
    override func encodeWithMorse(_ inputString: String) -> String? {
        return MorseEncoder.decode(inputString)
    }
    
    override func encodeWithURI(_ inputString: String) -> String? {
        return URLEncoder.decode(inputString)
    }
    
    override var encodeButtonTitle: String {
        return "Decode"
    }
}
